qiskit/compiler/schedule.py

```diff
@@ -40,6 +40,7 @@ def schedule(circuits: Union[QuantumCircuit, List[QuantumCircuit]],
              backend: Optional[BaseBackend] = None,
              inst_map: Optional[InstructionScheduleMap] = None,
              meas_map: Optional[List[List[int]]] = None,
+             dt: Optional[float] = None,
```

Comment:
Do we still need `dt`? I'm not sure it's actually used anymore?

Reply:
No, at least for this PR. I'm removing it from `schedule()` and `ScheduleConfig`.

Reply:
I've double checked and found we still need it when compiling a circuit with delays in seconds without backend. It might be a rare case but I think we need to care it, so I'll leave the current code here as it is.

Reply:
We assumed that `schedule` would only accept transpiled circuits without delay in seconds. I'm not too set on that, and we could discuss, but if you look at `lowering.py`:
```
        elif isinstance(inst, Delay):
            sched = Schedule(name=inst.name)
            for qubit in inst_qubits:
                for channel in [DriveChannel]:
                    sched += pulse_inst.Delay(duration=inst.duration, channel=channel(qubit))
            circ_pulse_defs.append(CircuitPulseDef(schedule=sched, qubits=inst_qubits))
```
This actually is not being taken care of --- the delay is instantiated with whatever the circuit duration is

Reply:
We're calling `delay_in_dt` before the above lines [-> here](https://github.com/Qiskit/qiskit-terra/blob/cbe7ac22046c6ed28435c59e64fa13cd81714e96/qiskit/scheduler/methods/lowering.py#L65) to convert delays in seconds to those in dts. I think it is working. I'm not sure where is the best place to call `delay_in_dt`, though. Maybe earlier stage in `schedule`?

Reply:
(This may be off topic...) I'd rather use the transpiler in the scheduler in the long run. I'm thinking `schedule` should just call `transpile` and `sequence` in the future. I tried to do that in this PR, but I ended up with leaving it for a future PR. I had difficulty in separating the logic to resolve the `meas_map` constraint, which is currently handled within `lowering.py`. (https://github.com/Qiskit/qiskit-terra/pull/4555#discussion_r448887371) IMO, `meas_map` should be taken in `transpile` rather than `sequence`/`lower_gates`.

Reply:
Anyway, there is no strong objection (i.e. no change request) here now, right?

Reply:
Can't this be in the `inst_map`? i.e. a circuit `delay` will map to a pulse `delay` with some unit conversion factor. Can we write a `InstructionScheduleMap` to capture such a thing? If so it should be populated automatically from backend (I know it is not right now).

With this we can also get rid of `delay_in_dt` I think (which I don't really like). The circuit should be able to contain any kind of delay (s, ns, dt). The `InstructionScheduleMap` should say how each of those map to pulses (which are exclusively in `dt`).

Reply:
It can be done in `inst_map`, however, that means we need to handle float durations at the scheduling stage. I prefer integer durations while scheduling so to be free from "allowable error margin". Thus I introduced `delay_in_dt` to convert all the durations into integer ones *before* scheduling.